src/qengine/opencl.cpp

```diff
@@ -869,14 +878,30 @@ void QEngineOCL::DecomposeDispose(bitLenInt start, bitLenInt length, QEngineOCLP
         partStateAngle[l] += angleOffset;
     }
 
+#if ENABLE_RASPBERRYPI
+        // The VC4CL implementation of sin() that the next kernel relies on appears to be bugged.
+        // Until this is fixed, we have to shunt the problem with a software implementation.
```

Comment:
How does the shunting happen?  Specifically, add a comment both here and (if I'm reading it right) below indicating where the shunt is, and - if possible - linking to a bug report on sin().

Reply:
We keep the probability and phase vectors mapped, so we can access them in "software." Then, we `LockSync()` so we can access the state vector in software, and we do what the kernel is supposed to do on the "host" side instead. Then, we unmap and `UnlockSync()` everything at the end.

I want to open an issue on the compiler project, but I'd like to give them an easy example. I'm getting results cos^2(x) + sin^2(x) >> 10, whereas it should always be (extremely close to) 1. This is true whether I use the vectorized sine or nonvectorized, switch the pi macros out, etc..

Reply:
See https://github.com/doe300/VC4CL/issues/54. I'll update the commentary, in the code.